modules/news.js

```diff
@@ -0,0 +1,225 @@
+//variable used to store the complete data of all news channels
+let news;
+window.onload = () => {
+	news = new News();
+};
+class News {
+	constructor() {
+		this.fullData = '';
+		this.displayPage();
+	}
+
+	//On Load Function Call
+	displayPage = () => {
+		this.header();
+		this.footer();
+		this.response();
+	};
+
+	header = () => {
+		let html = `<h1 class="header__heading">NEWSFEED</h1><p class="header__caption">Yet another newsfeed</p>
+				<input type='text' id='subscribe' class='header__subscribe-textbox' placeholder='Email Address'/>
+				<button class='header__subscribe-button' id='subBtn'>Subscribe</button>`;
+		document.getElementById('header').innerHTML = html;
+		document.getElementById('subBtn').addEventListener('click', validate);
+	};
+
+	footer = () => {
+		let html = `<p class="footer__copyright">&copy; NewsFeed 2019</p>`;
+		document.getElementById('footer').innerHTML = html;
+	};
+
+	async response() {
+		let url = 'https://newsapi.org/v2/top-headlines?' + 'country=us&' + 'apiKey=9fdb04ee4078412b82f9dd7f760464f8';
+		let req = new Request(url);
+		showLoader();
+		await fetch(req)
+			.then((res) => res.json()).then((data) => {
+				this.displayForm(data.articles);
+				this.displayNews(data.articles);
+			})
+			.catch(err => {
+				console.log(err);
+			});
+		closeLoader();
+	}
+
+	displayForm = (dataJSON) => {
+		let channels = [];
+		dataJSON.forEach((e) => {
+			channels.push(e.source.name);
+		});
+		//Remove duplicate channels using set operator
+		channels = [...new Set(channels)];
+
+		let formDivision = document.createElement('div');
+		formDivision.setAttribute('id', 'iamform');
+		let mainDivision = document.getElementById('main');
+		mainDivision.appendChild(formDivision);
+
+		let allChannels = '';
+		allChannels = channels.reduce(function (total, current) {
+			return total + `<option value='${current}'>${current}</option>`;
+		});
+
+		document.getElementById('iamform').innerHTML = `<div class='form'>
+				<label for='sel-category' class='form__select-label'><b>SELECT CATEGORY</b></label>
+				<select id='sel-category' class='form__select-box'>
+				${allChannels} 
+				</select>
+				</div>`;
+		document.getElementById('sel-category').addEventListener('change', function () {
```

Comment:
why not add onchange in the html itself

<select id='sel-category' class='form__select-box' onchange='selectCategoryNews(dataJSON)'>